### ISSUE TITLE
state is missing post finishHooks?

### ISSUE THREAD

**Issue by hemanth** [facebook/react#14004]
In the [example](https://codesandbox.io/s/xpkl0p52lq) invoking `loadImg` is resulting in `state` not being defined.

`XKCD` should become an `async` function so that we can `await loadImg` and wrap the `render` in an `async` function?

**Comment by HenriBeck** [comment]
No, the reason why `state` is not defined is that you need a default case for your reducer just to return the `state`.

See: https://codesandbox.io/s/5xpy0nvxyx

**Comment by hemanth** [comment]
@HenriBeck Dang! 

But what about the `loadImg` or `onLoad` (looks like this onload ain't triggering) ?

**Comment by HenriBeck** [comment]
Well, it's not triggering because the `src` property is an empty string.

**Comment by hemanth** [comment]
Yes, we do notice `<img src="">` but `data.state.comic.message` is a valid image URL.

**Comment by HenriBeck** [comment]
Yeah, but you aren't resetting the loading state in the `fetched` action which causes the image never to render.

**Comment by HenriBeck** [comment]
And the reason why the image is only showing briefly is that in the `img_fetched` action you are merging in again the `initialState` which causes `state.comic` to be an empty object again.
